### PR TITLE
Fix issue 22837 - checkConstructorEscape quits after first non-pointer

### PR DESCRIPTION
**Pull request by dkorpel**

### CODE REVIEW

**Comment by dkorpel** [comment]
Depends on https://github.com/dlang/phobos/pull/8398

File: test/fail_compilation/test19097.d [hunk 1]
```diff
@@ -1,7 +1,8 @@
 /* REQUIRED_ARGS: -preview=dip1000
  * TEST_OUTPUT:
 ---
-fail_compilation/test19097.d(35): Error: scope variable `s` may not be returned
+fail_compilation/test19097.d(36): Error: scope variable `s` may not be returned
+fail_compilation/test19097.d(74): Error: scope variable `s` may not be returned
```
Comment (MoonlightSentinel) [inline test/fail_compilation/test19097.d:5]: Shouldn't this error refer to `s.i` instead of `s`?

**Review by MoonlightSentinel — COMMENTED** [review]
> No description provided.

File: test/fail_compilation/test19097.d (hunk 1 above)
Comment (dkorpel) [reply test/fail_compilation/test19097.d:5]: escape.d likes to reduce expressions to 'the variable it really is', so s.x => s`, or `sarray[0] => sarray` etc. This is reflected in the error messages.

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**